src/app/copy/copy-service/copy.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Inject, Injectable } from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+
+  /**
+   * A generic service for copying text to clipboard
+   */
+@Injectable()
+export class CopyService {
+  public dom: Document;
+  private verbose: boolean = false;
+
+  /**
+   * The default constructor
+   */
+  constructor(@Inject(DOCUMENT) dom: Document) {
+    this.dom = dom;
+  }
+
+  /**
+   * Copy a value to the user's system clipboard
+   */
+  copy(value: string): boolean {
+    let result = false;
+    let textarea = this.dom.createElement('textarea');
+    const triggerElement = <HTMLElement>document.activeElement;
+
+    textarea.style.width = '0px';
+    textarea.style.height = '0px';
+    textarea.style.position = 'fixed';
+    textarea.style.top = '-100px';
+    textarea.style.left = '-100px';
+    textarea.style.opacity = '0';
+    textarea.value = value;
+
+    this.dom.body.appendChild(textarea);
+
+    textarea.select();
+
+    if (!!triggerElement) {
```

Comment:
Using `!!` is an obscure way to perform a type conversion. Please use one of the following approaches instead.

if (triggerElement !== undefined) {...} or
if (triggerElement) {...}